code/src/beetle.cpp

```diff
@@ -68,40 +65,124 @@ namespace r2d2::moving_platform {
         }
         // motor tests strats motors.
         qik_2s12v10_motorcontroller.set_m0_speed(motor_speed);
-        qik_2s12v10_motorcontroller.set_m1_speed(motor_speed);
+        qik_2s12v10_motorcontroller.set_m1_speed(-motor_speed);
 
         while (true && degrees != 0) {
             if (motor_encoder_m0.read() > adc_voltage) {
-                if (low_m0 == true) {
+                if (m0_state == false) {
                     counter_m0++;
                 }
-                low_m0 = false;
+                m0_state = true;
             } else {
-                low_m0 = true;
+                m0_state = false;
             }
-            if (counter_m0 ==
-                (int(encode_1_full_turn * turn / 360 * degrees))) {
+            if (counter_m0 == (uint_fast16_t(encode_1_full_turn * turn_factor /
+                                             360 * degrees))) {
                 qik_2s12v10_motorcontroller.brake_m0(20);
             }
 
             if (motor_encoder_m1.read() > adc_voltage) {
-                if (low_m1 == true) {
+                if (m1_state == false) {
                     counter_m1++;
                 }
-                low_m1 = false;
+                m1_state = true;
             } else {
-                low_m1 = true;
+                m1_state = false;
             }
-            if (counter_m1 ==
-                (int(encode_1_full_turn * turn / 360 * degrees))) {
+            if (counter_m1 == (uint_fast16_t(encode_1_full_turn * turn_factor /
+                                             360 * degrees))) {
                 qik_2s12v10_motorcontroller.brake_m1(20);
             }
-            if (counter_m0 > (encode_1_full_turn * turn / 360 * degrees) &&
-                counter_m1 > (encode_1_full_turn * turn / 360 * degrees)) {
+            if (counter_m0 >
+                    (encode_1_full_turn * turn_factor / 360 * degrees) &&
+                counter_m1 >
+                    (encode_1_full_turn * turn_factor / 360 * degrees)) {
                 break;
             }
-            // wait so the while loop aint blocking
-            hwlib::wait_ms(0.1);
+        }
+    }
+    void beetle_c::move(uint16_t distance) {
+        // Encode frequency for 1 turn of the wheel. the encoder has 64 point
+        // per over 2 pins we count when de adc goes from low to high of 1 pin.
+        uint_fast16_t encoder_rotations;
+        if (distance >= 10) {
```

Comment:
why is 10 the check you can only move 10  cm?

Reply:
The minimal moving distance is 10 cm

Reply:
this should be posted on the wiki somewhere